src/api/client.js

```diff
@@ -3,6 +3,13 @@ import Cookies from 'js-cookie';
 
 import endpoints from './endpoints';
 
+// terrible name, but it'll do for the moment
```

Comment:
The comment stands. I can't think of an elegant name to describe what this dictionary is for. Perhaps it's because the problem it attempts to solve is an ugly one.

Basically, the underlying assets database has the following names for asset attributes we care about: displayname, contentType, and uploadDate. Notice that displayname is not cased, contentType is camel-cased, and uploadDate is camel-cased but doesn't match the name returned by the API (date_added).

The API handles the conversion of date_added to uploadDate and display_name to displayname but not content_type to contentType.
The code for that is [here](https://github.com/edx/edx-platform/blob/master/cms/djangoapps/contentstore/views/assets.py#L279).

The actual API response you get [back](https://github.com/edx/edx-platform/blob/master/cms/djangoapps/contentstore/views/assets.py#L559) when requesting assets uses display_name, content_type, and date_added, so you can't just use the asset attribute names returned by the API to request a sort type, since the contentType case isn't handled. 

We could, alternatively, add a condition to the assets code to handle contentType, but I kind of just wanted to deal with it here so that it's completely explicit. I had to spend a good amount of time figuring out why this wasn't working, so I would like to avoid someone having to do the same later.